namespaces/cloud-platform-test-1.k8s.integration.dsd.io/kiam/README.md

```diff
@@ -0,0 +1,298 @@
+# Kiam - AWS IAM Role Management 
+
+## Overview
+
+**The purpose of this documentation is to serve the needs of all types of users that will interact with Kiam.**
+
+* Firstly, this document will guide anyone seeking to deploy Kiam on a new cluster for the first time.
+
+* Secondly, this document will educate anyone who simply would like to know how Kiam works and why we need it.
+
+* Lastly, the document will enable a user who is seeking to utilise Kiam on a cluster, where it has already been deployed and configured. 
+
+## What is Kiam, why we need it, and how it works.
+
+Traditionally in AWS, resources are allocated in the form of IAM Roles. An example of this would be if you had an instance running an application that needed to access an S3 bucket, you'd give it permission to do so via an IAM Policy within a Role and then you'd allow the instance to assume that Role.
+
+With the implementation of Kubernetes, we have entered into the world of Multi-Tenanted containers. As you will be aware, Nodes are still at their core, EC2 instances. Essentially, this means that the Nodes of a cluster will assume IAM credentials from AWS so the multiple containers in each Node can access the AWS resources they require to function.
+
+The problem with the above paragraph is that the IAM Role stops at the Node level and all the containers below the Nodes share the same credentials and permissions to all the AWS resources that are specified on the Nodes IAM Role. 
+
+An example of where this could become problematic is you may have an application in a cluster that requires full read and write access to a specific S3 Bucket. So to give access to the S3 bucket for the application, you'd have to modify the Node IAM Role to include a policy for full read and write access to S3. Bearing in mind, that all the containers under the Nodes share the same IAM role, you're not just given that singular container access to the S3 bucket, but every container under the Nodes.
```

Comment:
Someone could argue that instead of giving the policy to the _nodes role_ they might create a bunch of roles that the _nodes_ can assume. But of course any pod can assume any role the node can assume without kiam. This protects against that too.

Reply:
Feedback applied with Commit `c78bacf`